stdlib/knative/src/main/ballerina/src/knative/annotation.bal

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+# Knative annotation configuration.
+#
+#
+# # metadata for artifacts
+# Metadata for artifacts
+#
+# + name - Name of the resource. Default is `"<OUPUT_FILE_NAME>-<RESOURCE_TYPE>"`.
+# + labels - Map of labels for the resource.
+# + annotations - Map of annotations for resource.
+public type Metadata record {|
+    string name?;
+    map<string> labels?;
+    map<string> annotations?;
+|};
+# External file type for docker.
+#
+# + sourceFile - source path of the file (in your machine)
+# + target - target path (inside container)
+public type FileConfig record {|
+    string sourceFile;
+    string target;
+|};
+# Value for a field.
+#
+# + fieldPath - Path of the field
+public type FieldValue record{|
```

Comment:
Does this have to be a separate record? Given that it has only a single field?

Reply:
These records map to the kubernetes spec resources. Hence its cleaner to keep as this.

Reply:
Ack, +1.